### PR TITLE
Flesh out servers implementation

### LINKED ISSUE #220: Write a server module with a server summary
Should be very similar to the summary component code (#187) - can reuse bits of this.
Should also use the new Query module once #215 ships.

Part of #183

### PR DESCRIPTION
**Pull request by rmars**
Also add a dashboard-shared.css file for common styles.
Still holding out on styling this too much until all the js is done.
Colors coming in a branch by esbie.
Similar to esbie's banner code.

Fixes #220

screen shot 2016-04-01 at 12 21 25 pm

### CODE REVIEW

**Comment by klingerf** [comment]
:star: :dragon: Cool, looking good to me.

File: linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js [hunk 1]
```diff
@@ -1,28 +1,106 @@
...
+      {
+        description: "Failures",
+        metricSuffix: "failures",
+        query: Query.serverQuery().withRouter(routerName).withServer(serverName).withMetric("failures").build()
+      }
+    ];
+  }
+
+  function getServerName(server) {
+    // very temporary workaround for bug where server name has "requests" appended
```
Comment (esbie) [inline linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js]: sorry about that! should be fixed once you rebase/merge master

File: linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js [hunk 2]
```diff
@@ -1,28 +1,106 @@
...
+    // TODO: #198 remove
+    var name = server.label.split("/");
+    name.pop();
+    return name.join("/");
+  }
+
+  return function (metricsCollector, routers, $serverEl, routerName, serverTemplate) {
+    var servers = routers.servers(routerName);
+    var $serversContainer = $("<div class='router-servers clearfix'><div>Servers</div></div>");
```
Comment (esbie) [inline linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js]: I'd add this div directly to router_container.template

File: linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js [hunk 3]
```diff
@@ -1,28 +1,106 @@
...
+  function renderServer($container, server, data) {
+    var serverHtml = template({
+      server: server.label,
+      metrics: data
     });
-  }
+    var $serverHtml = ($("<div />").addClass("router-server").html(serverHtml));
 
-  function renderServers() {
+    $container.html($serverHtml);
```
Comment (esbie) [inline linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js]: I'm a little confused by this. If I'm understanding correctly, $container is a `<div class="router-server"/>` but you then stick another `<div class="router-server"/>` into it .... thereby recursively nesting this div on every update?

File: linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js [hunk 4]
```diff
@@ -1,28 +1,106 @@
...
     }
+
+    metricsCollector.registerListener(metricsHandler, getDesiredMetrics);
+
+    return {};
   };
 })();
+
+var RouterServers = (function() {
+  function getMetricDefinition(routerName, serverName) {
```
Comment (esbie) [inline linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js]: tioli, this method would probably work just as well inside `RouterServer` where it's actually used

Comment (rmars) [inline linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js]: yeah that does simplify things!

File: linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js (hunk 3 above)
Comment (rmars) [inline linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_server.js]: Fixed this... the intermediate div was unnecessary

**Comment by esbie** [comment]
:star: looks great, thanks for starting shared.css

**Pull request merged**